### PR TITLE
[GlobalISel][AArch64] Legalize G_INSERT_VECTOR_ELT for SVE

### PR DESCRIPTION
**Pull request by tschuett**
There are patterns for:
* {nxv2s32, s32, s64},
* {nxv4s16, s16, s64},
* {nxv2s16, s16, s64}

### CODE REVIEW

**Comment by tschuett** [comment]
Last PR had only MIR tests, which hid the LLVM-IR issue and made Windows CI unhappy.

File: llvm/test/CodeGen/AArch64/extract-vector-elt.ll [hunk 1]
```diff
@@ -143,11 +143,12 @@ define i64 @extract_v2i64_extract_build_vector_opaque(<2 x i64> %a, i32 %c) {
 ; CHECK-SD:       // %bb.0: // %entry
 ; CHECK-SD-NEXT:    sub sp, sp, #16
 ; CHECK-SD-NEXT:    .cfi_def_cfa_offset 16
-; CHECK-SD-NEXT:    adrp x8, .LCPI8_0
+; CHECK-SD-NEXT:    mov x8, #-31 // =0xffffffffffffffe1
```
Comment (arsenm) [inline llvm/test/CodeGen/AArch64/extract-vector-elt.ll:146]: Degrading test coverage in the non-SVE case?

Comment (tschuett) [reply llvm/test/CodeGen/AArch64/extract-vector-elt.ll:146]: This PR adds test-cases with scalable vectors, which are only supported *with* SVE.

Comment (arsenm) [reply llvm/test/CodeGen/AArch64/extract-vector-elt.ll:146]: Yes, but it is mixing tests with fixed vectors. Split the test for the scalable cases?

Comment (tschuett) [reply llvm/test/CodeGen/AArch64/extract-vector-elt.ll:146]: Do you want to move the new tests to a different file?

File: llvm/test/CodeGen/AArch64/extract-vector-elt.ll [hunk 2]
```diff
@@ -1099,5 +1099,3 @@ loop:
 ret:
   ret i32 %3
 }
-
```
Comment (arsenm) [inline llvm/test/CodeGen/AArch64/extract-vector-elt.ll:1102]: Leftover whitespace change

**Review by arsenm — APPROVED** [review]
lgtm with nit

File: llvm/test/CodeGen/AArch64/extract-vector-elt.ll (hunk 2 above)
Comment (tschuett) [reply llvm/test/CodeGen/AArch64/extract-vector-elt.ll:1102]: Trunk has no whitespace:
https://github.com/llvm/llvm-project/blob/main/llvm/test/CodeGen/AArch64/extract-vector-elt.ll

**Pull request merged**